src/code.cloudfoundry.org/silk/daemon/vtep/converger_test.go

```diff
@@ -2,437 +2,806 @@ package vtep_test
-					},
-					{ // not in overlay, skipped
-						UnderlayIP:          "10.10.0.3",
-						OverlaySubnet:       "10.254.11.0/24",
-						OverlayHardwareAddr: "aa:aa:00:00:00:01",
-					},
-					{ // not in overlay, skipped
-						UnderlayIP:          "10.10.0.4",
-						OverlaySubnet:       "10.254.12.0/24",
-						OverlayHardwareAddr: "aa:aa:00:00:00:02",
-					},
-					{ // in overlay
-						UnderlayIP:          "10.10.0.5",
-						OverlaySubnet:       "10.255.19.0/24",
-						OverlayHardwareAddr: "aa:aa:00:00:00:03",
-					},
-				}
+			Context("when deleting the route fails", func() {
+				BeforeEach(func() {
+					fakeNetlink.RouteDelReturns(errors.New("durian"))
+
+					destGW, destNet, _ := net.ParseCIDR("10.255.19.0/24")
+					fakeNetlink.RouteListReturns([]netlink.Route{
+						netlink.Route{
+							LinkIndex: 42,
+							Scope:     netlink.SCOPE_UNIVERSE,
+							Dst:       destNet,
+							Gw:        destGW,
+						},
+					}, nil)
+				})
+				It("returns a meaningful error", func() {
+					err := converger.Converge([]controller.Lease{})
+					Expect(err).To(MatchError("del route: durian"))
+				})
 			})
-			It("does not touch them and adds only the leases in the overlay network", func() {
-				err := converger.Converge(leases)
-				Expect(err).NotTo(HaveOccurred())
 
-				Expect(fakeNetlink.RouteReplaceCallCount()).To(Equal(1))
-				addedRoute := fakeNetlink.RouteReplaceArgsForCall(0)
-				Expect(addedRoute.Dst.IP).To(Equal(net.ParseIP("10.255.19.0").To4()))
-				Expect(fakeNetlink.NeighSetCallCount()).To(Equal(2))
-				addedARP := fakeNetlink.NeighSetArgsForCall(0)
-				Expect(addedARP.IP).To(Equal(net.ParseIP("10.255.19.0")))
-				addedFDB := fakeNetlink.NeighSetArgsForCall(1)
-				Expect(addedFDB.IP).To(Equal(net.ParseIP("10.10.0.5")))
-
-				Expect(logger.Logs()).To(HaveLen(1))
-				Expect(logger.Logs()[0].LogLevel).To(Equal(lager.INFO))
-				Expect(logger.Logs()[0].ToJSON()).To(MatchRegexp("converger.*non-routable-lease-count.*2"))
+			Context("when deleting a neigh fails", func() {
+				BeforeEach(func() {
+					fakeNetlink.NeighDelReturns(errors.New("mango"))
+
+					fakeNetlink.ARPListReturns([]netlink.Neigh{
+						netlink.Neigh{
+							LinkIndex:    42,
+							State:        netlink.NUD_PERMANENT,
+							Type:         syscall.RTN_UNICAST,
+							IP:           net.ParseIP("10.255.19.0"),
+							HardwareAddr: remoteMac,
+						},
+					}, nil)
+				})
+				It("returns a meaningful error", func() {
+					err := converger.Converge([]controller.Lease{})
+					Expect(err).To(MatchError("del neigh with ip/hwaddr 10.255.19.0 ee:ee:aa:aa:aa:ff: mango"))
+				})
 			})
-		})
 
-		Context("when a lease has an invalid MAC", func() {
-			BeforeEach(func() {
-				leases = []controller.Lease{
-					{
-						UnderlayIP:          "10.10.0.5",
-						OverlaySubnet:       "10.255.19.0/24",
-						OverlayHardwareAddr: "banana",
-					},
-				}
+			Context("when there are remote leases that are not in the overlay network", func() {
```

Comment:
what happens when the remote lease isn't in this daemon's lease's overlay network, but are in a different overlay network?